### PR TITLE
feat(progress): add warning variant

### LINKED ISSUE #4818: Add warning variation to progress component
https://github.com/patternfly/patternfly/pull/3423

A new variation for the progress component - `.pf-m-warning`

### PR DESCRIPTION
**Pull request by KKoukiou**
Closes #4818

### CODE REVIEW

**Comment by KKoukiou** [comment]
@redallen This seems to not be checking out the latests version of react-styles where this https://github.com/patternfly/patternfly/pull/3423/ is included. Thus CI is failing - on not finding the `pf-m-warning` class for the Progress component.
Should I do something to bump up the required version?

**Comment by redallen** [comment]
@KKoukiou You need to bump the version of `@patternfly/patternfly`. Check [npm](https://www.npmjs.com/package/@patternfly/patternfly) and then change it in one package (like react-styles) and run `node scripts/verififyPatternflyVersions --fix` -> `yarn install`.

File: packages/react-core/src/components/Progress/ProgressContainer.tsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import progressStyle from '@patternfly/react-styles/css/components/Progress/prog
 import { css } from '@patternfly/react-styles';
 import CheckCircleIcon from '@patternfly/react-icons/dist/js/icons/check-circle-icon';
 import TimesCircleIcon from '@patternfly/react-icons/dist/js/icons/times-circle-icon';
+import ExclamationTriangleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-circle-icon';
```
Comment (jenny-s51) [inline packages/react-core/src/components/Progress/ProgressContainer.tsx:6]: Change import path to '@patternfly/react-icons/dist/js/icons/exclamation-triangle-icon';`

**Review by jenny-s51 — COMMENTED** [review]
Thanks for working on this @KKoukiou ! Just change the icon to the exclamation triangle -- otherwise this is looking good!

File: packages/react-core/src/components/Progress/ProgressContainer.tsx (hunk 1 above)
Comment (KKoukiou) [reply packages/react-core/src/components/Progress/ProgressContainer.tsx]: Good catch! thanks

Comment (KKoukiou) [reply packages/react-core/src/components/Progress/ProgressContainer.tsx]: Oh the snapshot needs update as well. Will repost in 1h

**Review by jenny-s51 — APPROVED** [review]
LGTM 🚀

File: packages/react-core/src/components/Progress/examples/Progress.md [hunk 2]
```diff
@@ -62,6 +62,14 @@ import { Progress, ProgressSize, ProgressMeasureLocation, ProgressVariant } from
 <Progress value={33} title="Title" variant={ProgressVariant.danger} />
 ```
 
+### Warning
+```js
+import React from 'react';
+import { Progress, ProgressSize, ProgressMeasureLocation, ProgressVariant } from '@patternfly/react-core';
```
Comment (jschuler) [inline packages/react-core/src/components/Progress/examples/Progress.md:68]:
```suggestion
import { Progress, ProgressVariant } from '@patternfly/react-core';
```

**Review by jschuler — CHANGES_REQUESTED** [review]

**Review by jschuler — APPROVED** [review]

**Review by tlabaj — CHANGES_REQUESTED** [review]
Looks good.  Can you just update the demo app and integration test please.

**Comment by KKoukiou** [comment]
> Looks good. Can you just update the demo app and integration test please.

Added integration test - also for the other two missing variants - success and danger.

**Comment by tlabaj** [comment]
> > Looks good. Can you just update the demo app and integration test please.
> 
> Added integration test - also for the other two missing variants - success and danger.

Thank you!

**Review by tlabaj — APPROVED** [review]

**Review by jenny-s51 — APPROVED** [review]

**Pull request merged**